driver-core/src/main/com/mongodb/client/model/TimeSeriesOptions.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.model;
+
+import com.mongodb.lang.Nullable;
+
+import static com.mongodb.assertions.Assertions.notNull;
+
+/**
+ * Options related to the creation of time-series collections.
+ *
+ * @since 4.3
+ * @see CreateCollectionOptions
+ * @mongodb.driver.manual core/timeseries-collections/ Time-series collections
+ */
+public class TimeSeriesOptions {
+    private final String timeField;
+    private String metaField;
+    private TimeSeriesGranularity granularity;
+
+    /**
+     * Construct a new instance.
+     *
+     * @param timeField the name of the top-level field to be used for time. Inserted documents must have this field, and the field must be
+     *                 of the BSON datetime type.
+     */
+    public TimeSeriesOptions(final String timeField) {
+        this.timeField = notNull("timeField", timeField);
+    }
+
+    /**
+     * Gets the name of the field holding the time value.
+     *
+     * @return the name of the field holding the time value.
+     */
+    public String getTimeField() {
```

Comment:
We generally have not thought this was important to do in other options/settings classes, primarily because there is rarely a use case for the application to actually use the getters.  But I don't mind, I guess.

Done